apps/passport-server/test/devconnect.spec.ts

```diff
@@ -2910,35 +2910,71 @@ describe("devconnect functionality", function () {
     expect(unredactedTickets.length).to.eq(3);
   });
 
-  step("get poap claim urls from devconnect ticket ids", async () => {
-    // No POAP mint links in DB yet - all links return NULL
-    expect(await poapService.getDevconnectPoapClaimUrlByTicketId("1")).to.be
-      .null;
-    expect(await poapService.getDevconnectPoapClaimUrlByTicketId("2")).to.be
-      .null;
-
-    const TEST_POAP_LINK_1 = "https://poap.xyz/mint/qwerty";
-    const TEST_POAP_LINK_2 = "https://poap.xyz/mint/zxcvbn";
+  step(
+    "get poap claim urls from devconnect and zuzalu ticket ids",
+    async () => {
+      // No POAP mint links in DB yet - all links return NULL
+      expect(await poapService.getPoapClaimUrlByTicketId("1", "devconnect")).to
+        .be.null;
+      expect(await poapService.getPoapClaimUrlByTicketId("2", "devconnect")).to
+        .be.null;
+      expect(await poapService.getPoapClaimUrlByTicketId("3", "devconnect")).to
+        .be.null;
+      expect(await poapService.getPoapClaimUrlByTicketId("1", "zuzalu23")).to.be
+        .null;
+      expect(await poapService.getPoapClaimUrlByTicketId("2", "zuzalu23")).to.be
+        .null;
+      expect(await poapService.getPoapClaimUrlByTicketId("3", "zuzalu23")).to.be
+        .null;
+
+      const TEST_POAP_LINK_1 = "https://poap.xyz/mint/qwerty";
+      const TEST_POAP_LINK_2 = "https://poap.xyz/mint/zxcvbn";
+      const TEST_POAP_LINK_3 = "https://poap.xyz/mint/asdfgh";
+
+      await insertNewPoapUrl(db, TEST_POAP_LINK_1, "devconnect");
+      await insertNewPoapUrl(db, TEST_POAP_LINK_2, "zuzalu23");
+
+      // Map ticket ID "1" to a devconnect ticket
+      expect(
+        await poapService.getPoapClaimUrlByTicketId("1", "devconnect")
+      ).to.eq(TEST_POAP_LINK_1);
+      // Ran out of mint links for Devconnect
+      expect(await poapService.getPoapClaimUrlByTicketId("2", "devconnect")).to
+        .be.null;
 
-    await insertNewPoapUrl(db, TEST_POAP_LINK_1, "devconnect");
+      await insertNewPoapUrl(db, TEST_POAP_LINK_3, "devconnect");
 
-    expect(await poapService.getDevconnectPoapClaimUrlByTicketId("1")).to.eq(
-      TEST_POAP_LINK_1
-    );
-    // Ran out of mint links
-    expect(await poapService.getDevconnectPoapClaimUrlByTicketId("2")).to.be
-      .null;
-
-    await insertNewPoapUrl(db, TEST_POAP_LINK_2, "devconnect");
+      // Still maps to existing link, regardless of what the poapEvent parameter is
+      expect(
+        await poapService.getPoapClaimUrlByTicketId("1", "devconnect")
+      ).to.eq(TEST_POAP_LINK_1);
+      expect(
+        await poapService.getPoapClaimUrlByTicketId("1", "zuzalu23")
+      ).to.eq(TEST_POAP_LINK_1);
```

Comment:
I'm not sure I understand this behavior

> Still maps to existing link, regardless of what the poapEvent parameter is

why would you want that?

Reply:
The intended behavior is that the `poapEvent` parameter is only relevant when a new POAP mint link is being associated with a ticket ID. If an ticket ID is already associated with a POAP mint link, `poapEvent` should be irrelevant.

This was a specific choice given that the ticket ID was already the primary key in the table, so it wouldn't really make sense to also search an existing POAP mint link association by the key (ticketID, poapEvent). That being said, it's not a particularly strong opinion, and I can see arguments for the other direction.